include/zephyr/mem_mgmt/mem_attr_heap.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Carlo Caione, <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MEM_ATTR_HEAP_H_
+#define ZEPHYR_INCLUDE_MEM_ATTR_HEAP_H_
+
+/**
+ * @brief Memory heaps based on memory attributes
+ * @defgroup memory_attr_heap Memory heaps based on memory attributes
+ * @ingroup mem_mgmt
+ * @{
+ */
+
+#include <zephyr/mem_mgmt/mem_attr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Init the memory pool
+ *
+ * This must be the first function to be called to initialize the memory pools
+ * from all the memory regions with the attribute @ref DT_MEM_ALLOCATABLE
+ *
+ * @retval 0 on success.
+ * @retval -EALREADY if the pool was already initialized.
+ * @retval -ENOMEM too many regions already allocated.
+ */
+int mem_attr_heap_pool_init(void);
```

Comment:
This seems counterintuitive - shouldn't this just use the existing sys_init or device init systems?

I can see that it would be quite useful for testing though. Is that the intent?

Reply:
It is not. I just didn't want to force it down to the user if enabled. I can do that though.

Reply:
I guess there is no reason it couldn't be added in a follow-up commit.